sencha-workspace/SlateTasksManager/app/controller/Tasks.js

```diff
@@ -155,6 +159,13 @@ Ext.define('SlateTasksManager.controller.Tasks', {
         tasksStore.load();
     },
 
+    onIncludeUnsharedCheckboxClick: function(checkbox) {
+        var tasksStore = this.getTasksStore();
+
+        tasksStore.getProxy().extraParams.include_unshared = checkbox.checked;
+        tasksStore.load();
```

Comment:
Looks like if you're on the last page and then uncheck include shared you can end up stuck on an empty page. See if this fixes the problem by resetting the store back to the first page:

```suggestion
        tasksStore.load({ page: 1 });
```

Reply:
resolved: [9bd19ea](https://github.com/SlateFoundation/slate-cbl/pull/710/commits/9bd19ea543cca4730c594bcee0f294406ffa6b6c)